### PR TITLE
feat: [ANDROAPP-6392] Unify Date return values

### PR DESCRIPTION
**Pull request by xavimolloy**

### CODE REVIEW

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/internal/DateTimeUtils.kt [hunk 1]
```diff
@@ -7,13 +7,22 @@ import java.text.ParseException
 import java.text.SimpleDateFormat
 import java.util.Calendar
 
+@SuppressWarnings("deprecation")
+@Deprecated("This class is internal and should not be used replace with the new implementation")
 internal fun dateIsInRange(date: Long, allowedDates: SelectableDates, format: String = "ddMMyyyy"): Boolean {
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/internal/DateTimeUtils.kt:12]: if the class is for internal use, why deprecate it and not changing it?. It shouldn't provoke breaking changes, right?

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputDateTime.kt [hunk 2]
```diff
@@ -438,6 +455,381 @@ fun getSupportingTextList(uiModel: InputDateTimeModel, dateOutOfRangeItem: Suppo
...
+ * transformation.
+ * component uses Material 3 [DatePicker] and [TimePicker]
+ * input formats supported are mentioned in the date time input ui model documentation.
+ * [DatePicker] Input mode  will always follow locale format.
+ * @param uiModel: an [InputDateTimeModel] with all the parameters for the input
+ * @param modifier: optional modifier.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun InputDateTime(
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputDateTime.kt:529]: Can you create unit test to cover some examples of dates?

**Review by andresmr — CHANGES_REQUESTED** [review]

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputDateTime.kt (hunk 2 above)
Comment (xavimolloy) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputDateTime.kt:529]: will do!

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/internal/DateTimeUtils.kt (hunk 1 above)
Comment (xavimolloy) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/internal/DateTimeUtils.kt:33]: I have deprecated the one that is used by the old component although it won't create breaking changes it is no longer needed and I will remove it when we remove the deprecated component

**Review by andresmr — APPROVED** [review]

**Pull request merged**